components/base/sensorcontrolled/spin.go

```diff
@@ -46,11 +46,26 @@ func (sb *sensorBase) Spin(ctx context.Context, angleDeg, degsPerSec float64, ex
 		return err
 	}
 
+	// IsMoving returns true when moving, to determine if the spin call has started
+	baseMoving := func(ctx context.Context) (bool, error) {
+		moving, err := sb.IsMoving(ctx)
+		return moving, err
+	}
+
 	// IsMoving returns true when moving, which is not a success condition for our control loop
 	baseStopped := func(ctx context.Context) (bool, error) {
 		moving, err := sb.IsMoving(ctx)
 		return !moving, err
 	}
+
+	if err := sb.opMgr.WaitForSuccess(
+		ctx,
+		yawPollTime,
+		baseMoving,
+	); err != nil {
+		return err
+	}
+
```

Comment:
added some testing logs and this was the behavior I saw, so I do think the issue is that it was immediately returning before the motors had time to start moving
```
sensorcontrolled/spin.go:52	moving = false
sensorcontrolled/spin.go:52	moving = false
sensorcontrolled/spin.go:52	moving = true
sensorcontrolled/spin.go:59	moving = true
sensorcontrolled/spin.go:59	moving = true
```